### PR TITLE
add pre-commit script

### PR DESCRIPTION
**Pull request by yezhem**

### CODE REVIEW

**Review by merlintang — COMMENTED** [review]
can we have the precommit yaml file like this one:

https://github.com/couler-proj/couler/blob/master/.pre-commit-config.yaml

**Comment by yezhem** [comment]
it's enough to use the simple git hook script. 
pre-commit is suitable for large, multi-language projects.

File: .github/workflows/pre-commit [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+
+flake8 . --count --show-source --statistics --max-line-length=127 --max-complexity 15 --ignore=E722,W504
+
+pytest
```
Comment (merlintang) [inline .github/workflows/pre-commit:5]: why we need to add the pytest here?

**Review by mikecovlee — APPROVED** [review]
LGTM

**Pull request merged**